### PR TITLE
[PL-40539] [PL-40247] Reference JSON secrets

### PR DESCRIPTION
**Pull request by brian-f-harness**
Reference JSON secrets

For reviewers, previews are available:

[Add a custom secret manager](https://650d8c79da78164897326949--harness-developer.netlify.app/docs/platform/secrets/secrets-management/custom-secret-manager)
[Add an Azure Key Vault secret manager](https://650d8c79da78164897326949--harness-developer.netlify.app/docs/platform/secrets/secrets-management/azure-key-vault)
[Add a HashiCorp Vault secret manager](https://650d8c79da78164897326949--harness-developer.netlify.app/docs/platform/secrets/secrets-management/add-hashicorp-vault)
[Add an AWS secrets manager](https://650d8c79da78164897326949--harness-developer.netlify.app/docs/platform/secrets/secrets-management/add-an-aws-secret-manager)
[Add a Google Cloud secret manager](https://650d8c79da78164897326949--harness-developer.netlify.app/docs/platform/secrets/secrets-management/add-a-google-cloud-secret-manager)
[Reference existing secret manager secrets](https://650ddd52b7080319d8340088--harness-developer.netlify.app/docs/platform/secrets/secrets-management/reference-existing-secret-manager-secrets)

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/platform/secrets/shared/reference-via-json.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    "password": "db_P@ssw0rd!789"
+  }
+}
+
+```
+
+You can reference the secret path or use XPath for more complex JSON.
+
+```
+/<path>/user/database.password
```
Comment (ashishsanodia-harness) [inline docs/platform/secrets/shared/reference-via-json.md:21]: Format need to be fixed for referencing secret

To properly reference a secret, such as a JSON secret stored at `/<path>/user`, and if we want to access the password, we need to create a reference secret in the following format: /<path>/user#database.password.

File: docs/platform/secrets/secrets-management/custom-secret-manager.md [hunk 2]
```diff
@@ -23,11 +23,13 @@ This topic explains how to add and use a Custom Secret Manager in Harness.
 * Create/Edit Secrets
 * Create/Edit Connectors![](../../secrets/static/custom-secret-manager-31.png)
 
-### Important
+:::info important
 
-* Harness Custom Secret Manager is a read-only Secret Manager.
+* Harness Custom Secret Manager is a read-only Secret Manager. You cannot reference secrets in vault.
```
Comment (ashishsanodia-harness) [inline docs/platform/secrets/secrets-management/custom-secret-manager.md:28]: `You cannot reference secrets in vault.` is not correct

The custom secret manager is read-only, so creating reference secrets is not possible. Additionally, with a custom secret manager, users define templates that contain logic for connecting to and retrieving secrets from the secret manager. It's entirely up to the user's template to determine how the secret is read, whether it's in string or JSON format.

**Comment by ashishsanodia-harness** [comment]
@brian-f-harness I just checked the preview urls. looks like we need to update below section for reference secret
https://650d8c79da78164897326949--harness-developer.netlify.app/docs/platform/secrets/secrets-management/reference-existing-secret-manager-secrets#option-aws-secrets-manager-secrets

**Comment by brian-f-harness** [comment]
> @brian-f-harness I just checked the preview urls. looks like we need to update below section for reference secret https://650d8c79da78164897326949--harness-developer.netlify.app/docs/platform/secrets/secrets-management/reference-existing-secret-manager-secrets#option-aws-secrets-manager-secrets

Thank you @ashishsanodia-harness I added the content to the topic

**Review by ashishsanodia-harness — DISMISSED** [review]

File: docs/platform/secrets/shared/reference-via-json.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  "password": "P@ssw0rd!123",
+  "database": {
+    "username": "db_user",
+    "password": "db_P@ssw0rd!789"
+  }
+}
+
+```
+
+To reference a JSON secret and password stored in `/<path>/user`, you need to create a reference secret with the following format: `/<path>/user#database.password`.
```
Comment (deepakbolangady) [inline docs/platform/secrets/shared/reference-via-json.md:18]:
```suggestion
To reference a JSON secret and password stored in `/<path>/user`, create a reference secret with the following format: `/<path>/user#database.password`.
```

**Review by deepakbolangady — APPROVED** [review]

File: docs/platform/secrets/secrets-management/reference-existing-secret-manager-secrets.md [hunk 4]
```diff
@@ -16,7 +16,7 @@ Harness does not query the secret manager for existing secrets, but you can crea
 ### Before you begin
 
 * Go to [AWS KMS Secret Manager](/docs/platform/secrets/secrets-management/add-an-aws-kms-secrets-manager)
-* Go to [AWS Secrets Manager](/docs/platform/secrets/secrets-management/add-an-aws-secret-manager.md)
+* Go to [AWS Secret Manager](/docs/platform/secrets/secrets-management/add-an-aws-secret-manager.md)
```
Comment (brian-f-harness) [inline docs/platform/secrets/secrets-management/reference-existing-secret-manager-secrets.md:19]:
```suggestion
* Go to [AWS Secrets Manager](/docs/platform/secrets/secrets-management/add-an-aws-secret-manager.md)
```

**Pull request merged**